### ISSUE TITLE
Migrate modal to react-aria-components

### ISSUE THREAD

**Issue by MarekBodingerBA** [bratislava/konto.bratislava.sk#314]
Current modal solution doesn't use any library 😱 (focus trap doesn't work, questionable accesibility). We need to migrate to [react-aria-components](https://react-spectrum.adobe.com/react-aria/Modal.html). Not such a good example: https://github.com/bratislava/kupaliska-starz-fe/blob/e5e7bf1d14d41e4dce547d2820f2740292a1e882/src/components/Dialog/Dialog.tsx
